modules/ROOT/examples/live-demos/advtemplate-readonly-template/index.js

```diff
@@ -0,0 +1,347 @@
+    validate(content, isNonEmptyString, `Invalid "content" parameter: ${content}`)
+    const id = genId()
+    const template = {
+      id,
+      title,
+      content,
+    }
+    const parent = getParent(categoryId)
+    parent.items = [...parent.items, template]
+    templateIndex[id] = [template, categoryId];
+    return template
+  }
+
+  const renameCategory = (id, title) => {
+    validateTitle(title)
+    const category = getCategory(id)
+    category.title = title;
+  }
+
+  const renameTemplate = (id, title) => {
+    validateTitle(title)
+    const [template] = getTemplate(id)
+    template.title = title;
+  }
+
+  const deleteTemplate = (id) => {
+    const [, categoryId] = getTemplate(id)
+    const parent = getParent(categoryId)
+    parent.items = filterOut(parent.items, id)
+    delete templateIndex[id]
+  }
+
+  const deleteCategory = (id) => {
+    const category = getCategory(id);
+    category.items.forEach((template) => deleteTemplate(template.id))
+    store.items = filterOut(store.items, id)
+    delete categoryIndex[category.id]
+  }
+
+  const moveTemplate = (id, newCategoryId) => {
+    const [template, categoryId] = getTemplate(id)
+    if (categoryId === newCategoryId) {
+      return
+    }
+    const oldParent = getParent(categoryId)
+    const newParent = getParent(newCategoryId)
+    newParent.items = [...newParent.items, template]
+    oldParent.items = filterOut(oldParent.items, id)
+    templateIndex[id] = [template, newCategoryId]
+  }
+
+  const moveCategoryItems = (id, newCategoryId) => {
+    const oldCategory = getCategory(id)
+    if (id === newCategoryId) {
+      return
+    }
+    oldCategory.items.forEach((item) => moveTemplate(item.id, newCategoryId))
+  }
+
+  const load = (items, id) => {
+    for (const item of items) {
+      if (item.items) {
+        const category = createCategory(item.title, item.locked);
+        load(item.items, category.id );
+      } else {
+        createTemplate(item.title, item.content, id);
+      }
+    }
+  };
+
+  load(data);
+
+  return {
+    list,
+    getTemplate,
+    createCategory,
+    createTemplate,
+    renameCategory,
+    renameTemplate,
+    deleteTemplate,
+    deleteCategory,
+    moveTemplate,
+    moveCategoryItems,
+  }
+}
+
+
+const store = create(data);
+const advtemplate_list = async () => store.list()
+
+
+const advtemplate_create_category = async (title) => {
+  try {
+    return store.createCategory(title)
+  } catch (e) {
+    console.error(e)
+    throw new Error('Failed to create category')
+  }
+}
+  
+
+const advtemplate_create_template = async (title, content, categoryId) => {
+  try {
+    return store.createTemplate(title, content, categoryId)
+  } catch (e) {
+    console.error(e)
+    throw new Error('Failed to create template')
+  }
+}
+  
+
+const advtemplate_get_template = async (id) => {
+  try {
+    const [template ] = store.getTemplate(id)
+    return template
+  } catch (e) {
+    console.error(e)
```

Comment:
Missing a throw new error?

Reply:
Fixed in [this commit](https://github.com/tinymce/tinymce-docs/pull/3467/commits/5394480ac8a6dd555e21b3d1675dff757bd05f04)